airflow/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -1085,8 +1089,50 @@ def execute(self, context: Context) -> dict:
             raise AirflowException(f"Sagemaker Training Job creation failed: {response}")
 
         if self.deferrable and self.wait_for_completion:
+            description = self.hook.describe_training_job(self.config["TrainingJobName"])
+            status = description["TrainingJobStatus"]
+
+            if self.print_log:
+                instance_count = description["ResourceConfig"]["InstanceCount"]
+                last_describe_job_call = time.monotonic()
+                job_already_completed = status not in self.hook.non_terminal_states
+                _, last_description, last_describe_job_call = self.hook.describe_training_job_with_log(
+                    self.config["TrainingJobName"],
+                    {},
+                    [],
+                    instance_count,
+                    LogState.TAILING if job_already_completed else LogState.COMPLETE,
```

Comment:
Can we recheck if the conditional is correct here?

Reply:
Yes, i think the condition is correct. Is there any concern here?

Reply:
I was thinking if the condition should be other way round. Tail the logs while the job is not complete and if complete then set LogState.Complete.

Reply:
yep, thanks for the reminder. fixed it